### ISSUE TITLE
Optimize SVG files for smaller file size

### ISSUE THREAD

**Issue by MattSidor** [AlaskaAirlines/Icons#103]
**Is your feature request related to a problem? Please describe.**
SVG files generated by vector design programs such as Adobe Illustrator usually have a lot of room for optimization to reduce their file size. For example: they often contain unnecessary or redundant markup and metadata which can be removed; their vector paths can be recalculated to be simpler and more efficient while still tracing nearly identical lines; their output can be minified by consolidating to a single line.

By optimizing our Icon files, we can improve performance and reduce network bandwidth for the web pages and applications that consume them without degrading their image quality.

**Describe the solution you'd like**
I think we should add [SVGO](https://github.com/svg/svgo) to the build process for Auro Icons. It is a reputable and well-known node.js library for optimizing SVG files. By placing it in the build process, we may not need to worry as much about receiving ideal SVG markup from designers for new icons.

**Additional context**
[SVGOMG](https://jakearchibald.github.io/svgomg/) is a web-based app using SVGO that allows you to experiment with different settings to see how it changes the end result. The default settings in SVGOMG are the same as the default settings for SVGO. 

We would need to change the following default plugin settings to preserve the attributes we are setting in `icons.json`:
- `removeDesc: false` (removes `<desc>` when `true`)
- `removeTitle: false` (removes `<title>` when `true`)
- `removeUnusedNS: false` (removes `xmlns:xlink=` attribue when `true`)
- `removeUnknownsAndDefaults: { keepRoleAttr: true }` (removes `role=` when `false`)

A test with `payment/cc-alaska.svg` using SVGO with these settings reduced its file size from 61 KB to 23 KB.

**Comment by blackfalcon** [comment]
I am open to this idea. If you have already worked out a POC, I am open to seeing a PR.

**Comment by blackfalcon** [comment]
One thing I'd suggest, please do not turn on this optimization for all icons. The mono-color icons have already been optimized and we run a script to generate the final file from a data set. 

But the full-color icons like the credit cards, I agree there is some room to make those better.

**Comment by blackfalcon** [comment]
Ahh, discovered PR. It's a draft PR? Can you review the previous comment and let me know if we can do this?

**Comment by MattSidor** [comment]
re: draft PR — my bad, I misunderstood the contribution guidelines on the Auro homepage. I've marked it as ready for review.

re: differentiating mono-color icons — There is still a significant file size reduction on these icons after running through SVGO, so with all due respect, I would argue that it's still beneficial to run the library against these. I also think that it is best for this library to run on all icons agnostic of which icon it's processing and whatever work went into that icon ahead of time to optimize it manually before it reached the SVGO step.

An acceptance criteria for this change should be that the icon still looks the same and preserves all required attributes/tags such as "aria-", "role", and "<title>", and I think my PR achieves this even with the mono-color icons being run through this step.

In my PR, the SVGO `optimize()` function runs in the `generate.js` script on the `icon.svg` string as a final step before the string is written to disk.

**Comment by blackfalcon** [comment]
You are correct. "Once bitten, twice shy". The last time we tried to use an optimizer, it broke all the SVGs and they were unusable. I like how this is working in the generator pipeline. 

Like I said, the mono-color icons are pre-optimized, but the option this brings is to flatten the SVG HTML to a single line which amazingly removes close to 50% of the weight. 

This is a great issue and an awesome PR. Thanks for the submission.

**Comment by MattSidor** [comment]
Thanks Dale!

This also does change the icon's vectors by reducing their precision and complexity; just wanted to make sure you were aware of this before approving. That is probably the bulk of the weight reduction you are seeing.

**Example – error.svg**

***Before:***
`
<svg viewBox="0 0 24 24" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" role="img" aria-hidden="true" style="min-width: var(--auro-size-lg); height: var(--auro-size-lg); fill: currentColor" class="ico_squareLarge" >
    <title>Error</title>
    <desc>Error alert indicator</desc>
  <path d="M13.0467101,5.59878511 L19.83304,17.1845729 C20.3044044,17.9892977 19.7216697,19 18.7863299,19 L5.2136701,19 C4.27833028,19 3.69559564,17.9892977 4.16696005,17.1845729 L10.9532899,5.59878511 C11.420938,4.80040496 12.579062,4.80040496 13.0467101,5.59878511 Z M11.881817,6.46989182 C11.8464654,6.49025522 11.8170742,6.51940152 11.7965397,6.55445855 L5.41899252,17.4423625 C5.35372187,17.5537941 5.39190185,17.6965987 5.5042699,17.7613254 C5.54016625,17.7820026 5.58094009,17.7928939 5.62245286,17.7928939 L18.3775471,17.7928939 C18.5074965,17.7928939 18.6128413,17.688427 18.6128413,17.5595606 C18.6128413,17.5183937 18.6018584,17.4779597 18.5810075,17.4423625 L12.2034603,6.55445855 C12.1381897,6.44302691 11.9941851,6.4051651 11.881817,6.46989182 Z M12,15.5 C12.4142136,15.5 12.75,15.8357864 12.75,16.25 C12.75,16.6642136 12.4142136,17 12,17 C11.5857864,17 11.25,16.6642136 11.25,16.25 C11.25,15.8357864 11.5857864,15.5 12,15.5 Z M11.9764706,9.28 C12.301344,9.28 12.5647059,9.54116723 12.5647059,9.86333333 L12.5647059,14.2966667 C12.5647059,14.6188328 12.301344,14.88 11.9764706,14.88 C11.6515972,14.88 11.3882353,14.6188328 11.3882353,14.2966667 L11.3882353,9.86333333 C11.3882353,9.54116723 11.6515972,9.28 11.9764706,9.28 Z"/>
</svg>
`

After:
`
<svg viewBox="0 0 24 24" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" role="img" aria-hidden="true" style="min-width:var(--auro-size-lg);height:var(--auro-size-lg);fill:currentColor" class="ico_squareLarge"><title>Error</title><desc>Error alert indicator</desc><path d="m13.047 5.599 6.786 11.586A1.207 1.207 0 0 1 18.786 19H5.214a1.207 1.207 0 0 1-1.047-1.815l6.786-11.586a1.214 1.214 0 0 1 2.094 0Zm-1.165.87a.234.234 0 0 0-.085.085L5.419 17.442a.232.232 0 0 0 .203.35h12.756a.234.234 0 0 0 .203-.35L12.203 6.554a.236.236 0 0 0-.321-.084ZM12 15.5a.75.75 0 1 1 0 1.5.75.75 0 0 1 0-1.5Zm-.024-6.22c.325 0 .589.261.589.583v4.434a.586.586 0 0 1-.589.583.586.586 0 0 1-.588-.583V9.863c0-.322.264-.583.588-.583Z"/></svg>
`

**Comment by blackfalcon** [comment]
> This also does change the icons' vectors by reducing their precision and complexity

Yeah, I saw that. I reviewed a handful of the icons in your branch locally. This is a solid optimization tool. Thanks for finding the thing I didn't know existed! OSS for the WIN!
